### ISSUE TITLE
silently does not work if the static files folder path contains ".gz"

### ISSUE THREAD

**Issue by yarick123** [tkoenig89/express-static-gzip#26]
Hi,
I am trying to use the middleware express-static-gzip.

It seems, that I found a bug. If the name of the folder containing static gzipped files contains '.gz',
no pre-compressed content will be delivered.

Steps to reproduce:

1. create folder "wwwroot.gzipped" in the application root
2. put gzpped files into the folder, e.g. test.html.gz
3. register the folder as express-static-gzip root folder:
  app.use("/", expressStaticGzip('/wwwroot.gzipped'));
4. start the application, by me it works on localhost:8080
5. try to get ```http://localhost:8080/test.html```
6. result: error 404, "Cannot GET /test.html"

Renaming the folder to "wwwroot-gzipped" helps to avoid the problem.

The error is in index.js;167
```js
    /**
     * Adds the compression to the file's list of available compressions
     * @param {string} filePath
     * @param {Compression} compression
     */
    function addCompressionToFile(filePath, compression) {
// !! in the next line there is an error - fileExtension is removed also from the middle of filePath !!
        var srcFilePath = filePath.replace(compression.fileExtension, "").replace(rootFolder, "");
        var existingFile = files[srcFilePath];
        if (!existingFile) {
            files[srcFilePath] = { compressions: [compression] };
        } else {
            existingFile.compressions.push(compression);
        }
    }
```
So, in the line
```js
        var srcFilePath = filePath.replace(compression.fileExtension, "").replace(rootFolder, "");
```
fileExtension is removed also from the middle of filePath :(

I am using
- express-static-gzip version 1.1.1,
- node.js version 8.11.1,
- express version 4.16.3

Best regards
yarick123

**Comment by tkoenig89** [comment]
Should be resolved. I published version 1.1.2 to [npm](https://www.npmjs.com/package/express-static-gzip) which contains the fix.

**Comment by yarick123** [comment]
Thank you, the new version works in my environment with /wwwroot.gzipped/ folder
